src/uwtools/set_config.py

```diff
@@ -122,5 +132,14 @@ def create_config_obj(argv):
             out_object = config_obj
         out_object.dump_file(user_args.outfile)
 
+    if user_args.dry_run:
+        if user_args.outfile:
+            out_object = user_args.outfile
+            log.info(f'warning file {out_object} ',
+                 r"not written when using --dry_run")
+        # apply switch to allow user to view the results of config
+        # instead of writing to disk
+    log.info(config_obj)
+
```

Comment:
I think it could be good to move this logic up to just before the  `if user_args.outfile:` so that we don't do extra work if a user provides both an outfile and the dry-run flag.

Reply:
Oh shoot! I meant the whole if-statement -- lines 135 - 142 -- could move to up before line 118 in your version.